src/pinns_pde_solve.jl

```diff
@@ -658,7 +709,60 @@ function get_numeric_derivative()
         end
 end
 
+function get_numeric_integral(strategy, _indvars, _depvars, chain, derivative)
+    depvars,indvars,dict_indvars,dict_depvars = get_vars(_indvars, _depvars)
+    integral =
+        (u, cord, phi, integrating_var_id, integrand_func, lb, ub, θ ;strategy=strategy, indvars=indvars, depvars=depvars, dict_indvars=dict_indvars, dict_depvars=dict_depvars)->
+            begin
+                flat_θ = if (typeof(chain) <: AbstractVector) reduce(vcat,θ) else θ end
+                function integration_(cord, lb, ub, flat_θ)
+                    cord_ = cord
+                    function integrand_(x , p)
+                        @Zygote.ignore @views(cord_[integrating_var_id]) .= x
+                        return integrand_func(cord_, p, phi, derivative, nothing, u, nothing)
+                    end
+                    prob_ = QuadratureProblem(integrand_,lb, ub ,flat_θ)
+                    sol = solve(prob_,CubatureJLh(),reltol=1e-3,abstol=1e-3)[1]
+                    return sol
+                end
+                integration_arr = reshape([], 1, 0)
+                lb_definite, ub_definite = false , false
+                if (lb isa Number || lb isa Vector)
+                    lb_definite = true
+                end
+                if (ub isa Number || ub isa Vector)
+                    ub_definite = true
+                end
+                if !lb_definite
+                    lb_ = lb(cord , flat_θ, phi, derivative, nothing, u, nothing)
+                else
+                    if lb isa Number
+                        lb_ = fill(lb, 1, size(cord)[2])
+                    else
+                        lb_ = repeat(lb, 1, size(cord)[2])
+                    end
+                end
+                if !ub_definite
+                    ub_ = ub(cord , flat_θ, phi, derivative, nothing, u, nothing)
+                else
+                    if ub isa Number
+                        ub_ = fill(ub, 1 , size(cord)[2])
+                    else
+                        ub_ = repeat(ub, 1, size(cord)[2])
+                    end
+                end
+                for i in 1:size(cord)[2]
```

Comment:
if `cord = ones(2,300)`
so then 
julia> size(cord)[2]
300
why we calculate it for each index?  it is very slow

Reply:
in order to give the vectorise output. We have each index is a cord point in the data, so we need to calculate integral seperately for each point

Reply:
I did try `mapslices` but it threw a `Mutating arrays not supported` error

Reply:
it is a vector we do not need an index, we need a broadcast here for fast performance (see how it works in `get_numeric_derivative`).
In any case, it works fully correctly, but slow

Reply:
```julia
using BenchmarkTools

f(x) =x^2

function f_index(x)
    for i in 1:size(x)[2]
        x[:,i].^2
    end
end


julia> @benchmark f.(data) setup=(data=rand(2,1000))
BenchmarkTools.Trial: 10000 samples with 9 evaluations.
 Range (min … max):  2.525 μs …  3.818 ms  ┊ GC (min … max):  0.00% … 99.82%
 Time  (median):     3.732 μs              ┊ GC (median):     0.00%
 Time  (mean ± σ):   7.412 μs ± 85.571 μs  ┊ GC (mean ± σ):  28.22% ±  2.44%

  ▁ ▂ █▅▄▇▂ ▁                 ▃▄▅▄▄▄▃▃▃▃▂▂▂▁▁                ▂
  █▄█████████▆▅▄▅▄▅▆▅▅▅▃▅▆▅▅▅▇█████████████████▇▇▇▇▆▄▅▅▆▅▅▅▅ █
  2.52 μs      Histogram: log(frequency) by time     12.2 μs <

 Memory estimate: 15.75 KiB, allocs estimate: 1.

julia> @benchmark f_index(data) setup=(data=rand(2,1000))
BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):   56.758 μs … 36.600 ms  ┊ GC (min … max):  0.00% … 99.53%
 Time  (median):      78.680 μs              ┊ GC (median):     0.00%
 Time  (mean ± σ):   118.139 μs ±  1.100 ms  ┊ GC (mean ± σ):  29.35% ±  3.15%

  ▄▃ ▆█▆▆▆▄▅█▇▃▃▂▂▃▂▃▃▃▂▂▂▂▁▁▁▁▁▁▁▁▁  ▁▁                       ▂
  ████████████████████████████████████████▇▇█▇▇▇▇▇▇▆▆▆▇▆▆▄▅▄▅▅ █
  56.8 μs       Histogram: log(frequency) by time       188 μs <

 Memory estimate: 187.50 KiB, allocs estimate: 2000.
```